data-access/src/routes/http/graphql/schema/types/member.resolvers.ts

```diff
@@ -208,7 +208,42 @@ const member: Resolvers = {
         return { status: { success: true }, member };
       }
     },
-
+    memberUpdatePaymentInstrument: async (_, { input }, context) => {
+      const member = await getMemberForCurrentUser(context);
+      try {
+        let cybersourceCustomerId = member?.cybersourceCustomerId;
+        if (cybersourceCustomerId) {
+          const paymentInstrumentPayload: CustomerProfile = {
+            billingAddressLine1: input.billingAddressLine1,
+            billingAddressLine2: input.billingAddressLine2,
+            billingCity: input.billingCity,
+            billingState: input.billingState,
+            billingEmail: input.billingEmail,
+            billingFirstName: input.billingFirstName,
+            billingLastName: input.billingLastName,
+            billingPhone: input.billingPhone,
+            billingPostalCode: input.billingPostalCode,
+            billingCountry: input.billingCountry,
+            customerId: cybersourceCustomerId,
+          };
+          const paymentInfo: PaymentInstrumentInfo = {
+            id: input.id,
+            paymentInstrumentId: input.paymentInstrumentId,
+            cardExpirationMonth: input.expirationMonth,
+            cardExpirationYear: input.expirationYear,
+            cardType: input.cardType            
+          }
+          const response = await context.applicationServices.member.cybersourceApi.updatePaymentInstrument(paymentInstrumentPayload, paymentInfo);
+          if (response && input.isDefault) {
+            await context.applicationServices.member.cybersourceApi.setDefaultPaymentInstrument(cybersourceCustomerId, paymentInfo.paymentInstrumentId);
+            return { status: {success: true}, member };   
+          }
+          return { status: {success: false}, member };       
```

Comment:
Why this line is returning success: false here? What if someone updated payment instrument and did not ask for default change or vice versa?